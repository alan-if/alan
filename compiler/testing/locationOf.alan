Add To Every location
  Has existingAttribute 1.
End Add.

The l Isa location
End The l.


Start At l.
	  If location Of me Then "hej" End If. -- me undefined
	  Set location Of l To l. -- read-only attribute
	  Set location Of hero To l. -- read-only attribute
	  Set location Of hero To 4. -- read-only attribute & wrong type
	  If location Of hero = l Then "hopp" End If. -- OK
	  If existingAttribute Of location Of hero = 1 Then "Att = 1" End If. -- OK
	  If nonExistingAttribute Of location Of hero = 1 Then "Att = 1" End If. -- Non-existing attribute
