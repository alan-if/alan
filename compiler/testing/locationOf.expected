########## locationOf ##########


locationOf.alan

   10.  	  If location Of me Then "hej" End If. -- me undefined
=====>  	                 1

  *1*   310 E : Identifier 'me' not defined.

   11.  	  Set location Of l To l. -- read-only attribute
=====>  	               1

  *1*   436 E : Cannot modify readonly attribute.

   12.  	  Set location Of hero To l. -- read-only attribute
=====>  	               1

  *1*   436 E : Cannot modify readonly attribute.

   13.  	  Set location Of hero To 4. -- read-only attribute & wrong type
=====>  	  1            2

  *1*   331 E : Incompatible types in target and expression in SET statement.
  *2*   436 E : Cannot modify readonly attribute.

   16.  	  If nonExistingAttribute Of location Of hero = 1 Then "Att = 1" End I
        f. -- Non-existing attribute
=====>  	     1

  *1*   317 E : Attribute 'nonExistingAttribute' is not defined for the
                expression since the class it is guaranteed to be in this
                context ('location') does not have it.


        6 error(s).
        No detected warnings.
        1 informational message(s).

